### PR TITLE
[bitnami/fluentd] Make servicemonitor URL/Path configurable

### LINKED ISSUE #13388: [bitnami/fluentd] Make servicemonitor URL/Path configurable
### Name and Version

bitnami/fluentd

### What is the problem this feature will solve?

GIves the user the ability to to scrape aggregated_metrics via Prometheus

### What is the feature you are proposing to solve the problem?

Currently, the metrics path is hard coded and would like to make it configurable in servicemonitor.yaml

Hardcoded
`spec:endpoints:path: /metrics`

Please make path configurable

### What alternatives have you considered?

Overriding the fluentd.conf files /aggregated_metrics path with /metrics

### PR DESCRIPTION
**Pull request by cmbaatz**
### Description of the change
Adds logic that allows a user to override both the port and path used in the prometheus ServiceMonitor. <br/>
This change allows for the port & path configuration for the ServiceMonitor to be configured in two different ways.
1. Via dedicated values<br/>
    - `.Values.metrics.service.port`: Already defines the port for the service.  Reusing this value prevents a chart user for accidentally using two separate values.
    -  `.Values.metrics.serviceMonitor.path`: Defines the path that prometheus should scrape.  
2. Via legacy annotations<br/>
    Supports the legacy non-operator prometheus deployment in which these values would typically be set via annotations.  This will aid legacy users if they transition over to the new operator prometheus.<br/>
    - .Values.metrics.service.annotations.prometheus.io/port 
    - .Values.metrics.service.annotations.prometheus.io/path

### Benefits
Allows a user to change both the port and path used by prometheus.  

### Possible drawbacks
N/A

### Applicable issues
fixes #13388 

### Additional information
N/A

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/fluentd/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -26,8 +26,8 @@ spec:
       {{- include "common.tplvalues.render" (dict "value" .Values.metrics.serviceMonitor.selector "context" $) | nindent 6 }}
       {{- end }}
   endpoints:
-    - port: tcp-metrics
-      path: "/metrics"
+    - port: {{ default "tcp-metrics" (default (index .Values.metrics.service.annotations "prometheus.io/port") .Values.metrics.service.port) | quote}}
```
Comment (javsalgar) [inline bitnami/fluentd/templates/servicemonitor.yaml:29]: is it necessary to override the tcp-metrics string by `metrics.service.port`? In principle this should be the same, right?

Comment (cmbaatz) [reply bitnami/fluentd/templates/servicemonitor.yaml]: Fixed . thanks for the suggestion

File: bitnami/fluentd/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -26,8 +26,8 @@ spec:
       {{- include "common.tplvalues.render" (dict "value" .Values.metrics.serviceMonitor.selector "context" $) | nindent 6 }}
       {{- end }}
   endpoints:
-    - port: tcp-metrics
-      path: "/metrics"
+    - port: "tcp-metrics"
+      path: {{ default "/metrics" (default (index .Values.metrics.service.annotations "prometheus.io/path") .Values.metrics.serviceMonitor.path) | quote}}
```
[the remark this answers is not in the document]
Comment (cmbaatz) [reply bitnami/fluentd/templates/servicemonitor.yaml]: @javsalgar  actually after switching up to coalesce I realized that this wont work due to the fact I'm setting /metrics as the default value for .Values.metrics.serviceMonitor.path in values.yaml.  Therefore I switched by to using default, but now it only one statement first checking if the annotation is there and then defaulting to what's defined specified value.

**Comment by cmbaatz** [comment]
@javsalgar or @CeliaGMqrz is it possible to move this pull request forward?

**Review by CeliaGMqrz — DISMISSED** [review]
Sorry for the delay, LGTM! Thanks

**Review by CeliaGMqrz — APPROVED** [review]

**Pull request merged**